### DISCUSSION TITLE
Unable to fetch all the images from the PDF

### DISCUSSION THREAD

**Discussion by krish-tech02** [pymupdf/PyMuPDF#4061] [Looking for help] [answered]
@JorjMcKie Could you please help here? 
Unable to fetch all the images from the PDF. I am attaching the sample PDF, there is image in the every page starting of the PDF  "Advocate Health Care Aurora Health Care" but it is not getting fetched. I am using code to get images of the page:
 for img_index, img in enumerate(page.get_images(full=True)):
            print(f'img index: {img_index}')
            xref = img[0]  
            base_image = pdf_document.extract_image(xref)
            image_bytes = base_image["image"]
            image_ext = base_image["ext"]

[Headaches and Migraines.111524.MacOfc365.pdf](https://github.com/user-attachments/files/17805394/Headaches.and.Migraines.111524.MacOfc365.pdf)

**Comment by JorjMcKie** [comment]
There are exactly 2 images in the PDF, one on page 0 and one on page 1. They also get extracted.
So everything is fine!

**Reply by krish-tech02** [reply]
@JorjMcKie What about the header in every page is it not a image? attaching the screenshot for reference:
![image](https://github.com/user-attachments/assets/077080d5-79db-4a7c-89e2-d4814336cbc4)

**Reply by krish-tech02** [reply]
@JorjMcKie Could you please clarify if the content mentioned above consists of images? If it's not an image, is there a way to retrieve this content?

**Reply by JorjMcKie** [reply]
No, they are no images, but vector graphics.

**Reply by krish-tech02** [reply]
@JorjMcKie Thanks for the confirmation, is there any way to fetch graphics?

**Reply by JorjMcKie** [reply]
You can try and let method `page.cluster_drawings()` identify vector graphics that "belong together" (are close enough ...).
The returned rectangles can then be used as clips in `page.get_pixmap`. Which then can be save as images.
I have marked the default output here in red:
![image](https://github.com/user-attachments/assets/b6446c56-1847-41ed-ae01-329fe018fcd4)

**Reply by krish-tech02** [reply]
Sure, will try that, Thank you @JorjMcKie

**Reply by JorjMcKie** [reply, marked as answer]
Please make sure to read the documentation of `.cluster_drawings()`. You can influence its algorithm in some ways. If you set `x_tolerance=10` (default is 3), then fewer separate rectangles may come out, e.g. the blue ones here:
![image](https://github.com/user-attachments/assets/714eebd6-a8d3-4a03-8d63-632c779d216a)
